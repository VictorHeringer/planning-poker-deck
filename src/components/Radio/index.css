.radio {
  height: 10px;
  width: 10px;
  border-radius: 50%;
  float: right;
  position: relative;
  margin-top: 5px;
  cursor: pointer;
}

.radio.active {
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

.radio.inactive {
  -webkit-transition: all 0.4s;
  transition: all 0.4s;
}

/* Tech them */
.radio.tech-theme.active {
  background-color: #66BB6A;
}

.radio.tech-theme.inactive {
  background-color: #E0E0E0;
}

/* Dark them */
.radio.dark-theme.active {
  background-color: #212121;
}

.radio.dark-theme.inactive {
  background-color: #E0E0E0;
}

/* Eletric them */
.radio.eletric-theme.active {
  background-color: #F06292;
}

.radio.eletric-theme.inactive {
  background-color: #E0E0E0;
}