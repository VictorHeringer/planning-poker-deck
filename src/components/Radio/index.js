import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

/**
 * @author Victor Heringer
 * 
 * Radio component
 * 
 * @prop {String} id 
 * @prop {bool} active
 * @prop {Function} onClick
 */
const Radio = ({ id, active, onClick }) => {
  const activeClass = active ? 'active radio' : 'inactive radio';
  return <div className={activeClass} onClick={() => onClick(id)}></div>;
}

Radio.propTypes = {
  id: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool.isRequired
};

export default Radio;