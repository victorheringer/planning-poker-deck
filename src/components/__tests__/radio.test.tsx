import { render } from "@testing-library/react";
import Radio from "../radio";

describe("Radio", () => {
  it("should render correctly active radio", () => {
    const tree = render(<Radio active />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  it("should render correctly not active radio", () => {
    const tree = render(<Radio active={false} />).asFragment();
    expect(tree).toMatchSnapshot();
  });
});
