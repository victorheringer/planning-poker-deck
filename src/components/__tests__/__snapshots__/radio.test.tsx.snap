// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render correctly active radio 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cUlkLJ"
  />
</DocumentFragment>
`;

exports[`Radio should render correctly not active radio 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL cUlkLJ"
  />
</DocumentFragment>
`;
